hbase_sql/src/test/java/com/splicemachine/derby/impl/sql/execute/operations/CreateTableTypeHelper.java

```diff
@@ -0,0 +1,243 @@
+    static public int[] getParquetTypes() {
+        return new int[]{
+                Types.VARCHAR, Types.CHAR,
+                Types.DATE, Types.TIMESTAMP,
+                // Types.TIME, // supported, but will be written as TIMESTAMP
+                Types.TINYINT, Types.SMALLINT, Types.INTEGER, Types.BIGINT,
+                Types.DOUBLE, Types.REAL, Types.DECIMAL, Types.BOOLEAN
+        };
+    }
+
+    /// @return types that are supported by ORC
+    static public int[] getORCTypes() {
+        return new int[]{
+                Types.VARCHAR, Types.CHAR,
+                Types.DATE, Types.TIMESTAMP,
+                // Types.TIME, // supported, but will be written as TIMESTAMP
+                Types.TINYINT, Types.SMALLINT, Types.INTEGER, Types.BIGINT,
+                Types.DOUBLE, Types.REAL, Types.DECIMAL, Types.BOOLEAN
+        };
+    }
+
+    /// @return types that are supported by Avro
+    static public int[] getAvroTypes() {
+        return new int[]{
+                Types.VARCHAR, Types.CHAR,
+                // Types.DATE, Types.TIME // not supported
+                Types.TIMESTAMP,
+                //Types.TINYINT, Types.SMALLINT, // not supported
+                Types.INTEGER, Types.BIGINT,
+                Types.DOUBLE, Types.REAL,
+                // Types.DECIMAL, // not supported
+                Types.BOOLEAN
+        };
+    }
+
+    /// @return supported types by fileFormat PARQUET, ORC or AVRO
+    static public int[] getTypes(String fileFormat) {
+        if(fileFormat.equalsIgnoreCase("PARQUET"))
+            return getParquetTypes();
+        else if(fileFormat.equalsIgnoreCase("ORC"))
+            return getORCTypes();
+        else if(fileFormat.equalsIgnoreCase("AVRO"))
+            return getAvroTypes();
+        throw new RuntimeException("unsupported fileformat " + fileFormat);
+    }
+
+    /// compare that result in ResultSet rs is the same as from the generated insert values.
+    public void checkResultSetSelectAll(ResultSet rs) throws SQLException {
+        ArrayList<String> results = new ArrayList<>();
+        int nCols = rs.getMetaData().getColumnCount();
+        while (rs.next()) {
+            StringBuilder sb = new StringBuilder();
+            for (int i = 1; i <= nCols; i++) {
+                Object value = rs.getObject(i);
+                if (value != null && value instanceof Clob) {
+                    throw new RuntimeException("Clob not supported");
+                } else {
+                    if( i > 1 ) sb.append(", ");
+                    if( value != null && value.toString() != null ) {
+                        sb.append("'");
+                        sb.append(value.toString());
+                        sb.append("'");
+                    }
+                    else {
+                        sb.append("NULL");
+                    }
+                }
+            }
+            results.add(sb.toString());
+        }
+        results.sort(String::compareTo);
```

Comment:
I guess this could also be replaced by streams, but that's not blocking

Reply:
ResultSet doesn't have streaming interface, would need 1-2 helper classes for this.